elements/lisk-validator/src/formatter.ts

```diff
@@ -29,95 +30,122 @@ import {
-
-validator.addFormat('nonTransferAmount', validateNonTransferAmount);
-
-validator.addFormat('fee', validateFee);
-
-validator.addFormat('emptyOrPublicKey', data => {
-	if (data === null || data === '') {
-		return true;
+class LiskValidator {
+	private readonly validator: Ajv.Ajv;
+	public constructor() {
+		this.validator = new Ajv({
+			allErrors: true,
+			schemaId: 'auto',
+			useDefaults: false,
+		});
+
+		this.validator.addFormat('signature', isSignature);
+
+		this.validator.addFormat(
+			'id',
+			data =>
+				isNumberString(data) &&
+				!isGreaterThanMaxTransactionId(new BigNum(data)),
+		);
+
+		this.validator.addFormat('address', data => {
+			try {
+				validateAddress(data);
+
+				return true;
+			} catch (error) {
+				return false;
+			}
+		});
+
+		this.validator.addFormat('amount', isNumberString);
+
+		this.validator.addFormat('transferAmount', validateTransferAmount);
+
+		this.validator.addFormat('nonTransferAmount', validateNonTransferAmount);
+
+		this.validator.addFormat('fee', validateFee);
+
+		this.validator.addFormat('emptyOrPublicKey', data => {
+			if (data === null || data === '') {
+				return true;
+			}
+
+			try {
+				validatePublicKey(data);
+
+				return true;
+			} catch (error) {
+				return false;
+			}
+		});
+
+		this.validator.addFormat('publicKey', data => {
+			try {
+				validatePublicKey(data);
+
+				return true;
+			} catch (error) {
+				return false;
+			}
+		});
+
+		this.validator.addFormat('signedPublicKey', data => {
+			try {
+				const action = data[0];
+				if (action !== '+' && action !== '-') {
+					return false;
+				}
+				const publicKey = data.slice(1);
+				validatePublicKey(publicKey);
+
+				return true;
+			} catch (error) {
+				return false;
+			}
+		});
+
+		this.validator.addFormat('additionPublicKey', data => {
+			const action = data[0];
+			if (action !== '+') {
+				return false;
+			}
+			try {
+				const publicKey = data.slice(1);
+				validatePublicKey(publicKey);
+
+				return true;
+			} catch (error) {
+				return false;
+			}
+		});
+
+		this.validator.addFormat('username', isUsername);
+
+		this.validator.addFormat(
+			'noNullCharacter',
+			data => !isNullCharacterIncluded(data),
+		);
+
+		this.validator.addKeyword('uniqueSignedPublicKeys', {
+			type: 'array',
+			compile: () => (data: ReadonlyArray<string>) =>
+				new Set(data.map((key: string) => key.slice(1))).size === data.length,
+		});
+
+		this.validator.addFormat('hex', isHexString);
+
+		this.validator.addFormat('csv', isCsv);
 	}
 
-	try {
-		validatePublicKey(data);
-
-		return true;
-	} catch (error) {
-		return false;
-	}
-});
-
-validator.addFormat('publicKey', data => {
-	try {
-		validatePublicKey(data);
-
-		return true;
-	} catch (error) {
-		return false;
-	}
-});
-
-validator.addFormat('signedPublicKey', data => {
-	try {
-		const action = data[0];
-		if (action !== '+' && action !== '-') {
-			return false;
+	public validate(schema: object, data: object): [] {
```

Comment:
You can also create a method `validateAndThrow` or pass third param to `validate` to throw instead of return. This will simplify the usage in other code files. Just a suggestion.

Reply:
On Friday we discussed that the library should not throw and I do agree with that if you think it should throw or an alternate method should throw please issue a new PR against development.